unified-scheduler-pool/src/lib.rs

```diff
@@ -3014,17 +3082,53 @@ mod tests {
         let bank = Bank::new_for_tests(&genesis_config);
         let (bank, _bank_forks) = setup_dummy_fork_graph(bank);
         let ignored_prioritization_fee_cache = Arc::new(PrioritizationFeeCache::new(0u64));
-        let pool = SchedulerPool::<PooledScheduler<StallingHandler>, _>::new_dyn(
+        let pool = SchedulerPool::<PooledScheduler<StallingHandler>, _>::new(
             None,
             None,
             None,
             None,
             ignored_prioritization_fee_cache,
         );
-        let context = SchedulingContext::for_verification(bank.clone());
+        let (ledger_path, _blockhash) = create_new_tmp_ledger_auto_delete!(&genesis_config);
+        let blockstore = Arc::new(Blockstore::open(ledger_path.path()).unwrap());
+        let leader_schedule_cache = Arc::new(LeaderScheduleCache::new_from_bank(&bank));
+        let (_banking_packet_sender, banking_packet_receiver) = crossbeam_channel::unbounded();
+        let (exit, poh_recorder, poh_service, _signal_receiver) =
+            create_test_recorder_with_index_tracking(
+                bank.clone(),
+                blockstore.clone(),
+                None,
+                Some(leader_schedule_cache),
+            );
 
-        assert_eq!(bank.transaction_count(), 0);
+        if matches!(scheduling_mode, BlockProduction) {
+            pool.register_banking_stage(
+                banking_packet_receiver,
+                Box::new(|_, _| unreachable!()),
+                poh_recorder.read().unwrap().new_recorder(),
+            );
+        }
+
+        // Wait until genesis_bank reaches its tick height...
+        while poh_recorder.read().unwrap().bank().is_some() {
+            sleep(Duration::from_millis(100));
+        }
```

Comment:
best to have some timeout imo so this doesn't loop indefinitely.

Reply:
do you specifically need tiem to pass here? or can you warp the bank & set poh  manually?